src/main/java/seedu/address/model/person/Person.java

```diff
@@ -17,16 +18,16 @@
 public class Person {
 
     // Identity fields
-    private final Name name;
+    private Name name;
 
     // Data fields
-    private final Map<String, Phone> numbers;
-    private final Map<String, Email> emails;
-    private final Map<String, Address> addresses;
-    private final Company company;
-    private final JobTitle jobTitle;
-    private final HashSet<Pronoun> pronouns = new HashSet<>();
-    private final HashSet<Tag> tags = new HashSet<>();
+    private Map<String, Phone> numbers;
+    private Map<String, Email> emails;
+    private Map<String, Address> addresses;
+    private Company company;
+    private JobTitle jobTitle;
+    private HashSet<Pronoun> pronouns = new HashSet<>();
+    private HashSet<Tag> tags = new HashSet<>();
```

Comment:
If we are trying to preserve immutability, how about instead of creating a copy of a person, we just have getNumbers, getAddresses, etc. return immutable maps? Then we don't need to go through the hassle of creating a copy of a person just to do what we can already do with the original person